transport_plugins/native_ble/iotile_transport_native_ble/tilebus.py

```diff
@@ -0,0 +1,54 @@
+from bable_interface import Characteristic, Service
+import uuid
+
+# Company ID
+ArchManuID = 0x03C0
+
+# GATT table
+BLEService = Service(uuid='1800', handle=0x0001, group_end_handle=0x0005)
```

Comment:
Are these handles and `group_end_handle` only used on the virtual interface?  If so that's totally fine but if you key in on the handles for PODs as well that will break in the future as soon as we tweak a ble characteristic on the POD firmware and the handles are reassigned.  In that case I wouldn't fix it now but file an issue to we remember to fix it in the future when we make such a change, which won't be too soon.

Reply:
It is just that currently bable_interface.Service object are not linked to bable_interface.Characteristic: there is no hierarchy because it was quite annoying to serialize/unserialize then. But it is definitely something to change in baBLE: add an `add_characteristic` method to add a characteristic to a service and then, automatically calculate the `group_end_handle`, and the Characteristics `handle`,`value_handle`, `config_handle`.

This information is needed to register the GATT table so I need them only in the virtual interface.

So yes for now, we'll have to keep this data the same as the GATT table registered in the PODs. I'll create an issue

Reply:
After thinking about that, we'll always have to keep this updated when modifying the GATT table in the POD firmware, just like we have to in the BLED112 transport plugin (cf [virtual_bled112.py#L50](https://github.com/iotile/coretools/blob/master/transport_plugins/bled112/iotile_transport_bled112/virtual_bled112.py#L50)). 

So even if adding the Service/Characteristic hierarchy would be great (cf [issue here](https://github.com/iotile/baBLE/issues/23)), it won't solve this issue.